src/Components/Activities/ActivitiesForm.js

```diff
@@ -1,48 +1,202 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import Axios from 'axios';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import TextField from '@mui/material/TextField';
+import Card from '@mui/material/Card';
+import Button from '@mui/material/Button';
+import CardHeader from '@mui/material/CardHeader';
+import { Alert } from '@mui/material';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { useDropzone } from 'react-dropzone';
+import Swal from 'sweetalert2';
 import '../FormStyles.css';
+import '../../Styles/ActivitiesForm.css';
 
-const ActivitiesForm = () => {
+const thumb = {
+  display: 'inline-flex',
+  borderRadius: 2,
+  border: '1px solid #eaeaea',
+  margin: 'auto',
+  width: 'auto',
+  height: 100,
+  padding: 4,
+  boxSizing: 'border-box',
+};
+
+const ActivitiesForm = ({ id }) => {
   const [initialValues, setInitialValues] = useState({
-    name: '',
-    description: '',
+    name: id ? id.data.name : '',
+    description: id ? id.data.description : '',
+    image: id ? id.data.name : null,
   });
 
-  const handleChange = (e) => {
-    if (e.target.name === 'name') {
-      setInitialValues({ ...initialValues, name: e.target.value });
-    }
-    if (e.target.name === 'description') {
-      setInitialValues({ ...initialValues, description: e.target.value });
-    }
-  };
+  const [files, setFiles] = useState([]);
+  const { getRootProps, getInputProps } = useDropzone({
+    multiple: false,
+    maxFiles: 1,
+    accept: 'image/jpeg, image/png',
+    onDrop: (acceptedFiles) => {
+      setInitialValues({
+        ...initialValues,
+        image: acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      });
```

Comment:
what is being done inside of setInitialValues, image? The answer of that question is a good candidate for the name of the function where this peace of code should be. Please move this to a separated function and then call it here.
```
const getImageWithUrl= (acceptedFiles) => acceptedFiles.map((file) =>
          Object.assign(file, {
            preview: URL.createObjectURL(file),
          }),
        );
```
then on the onFileDrop fuction
```
setInitialValues({...initialValue, image: getFilesWithUrl(acceptedFiles)})
```